### PR TITLE
Asciidoc for scheduler

### PR DESCRIPTION
**Pull request by jayunit100**
Add more structure to the scheduler introduction.  This diagram should be easier to maintain as we add more features.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31612)

### CODE REVIEW

**Comment by jayunit100** [comment]
cc @kubernetes/sig-scheduling

File: docs/devel/scheduler.md [hunk 1]
```diff
@@ -41,45 +41,67 @@ indicating where the Pod should be scheduled.
...
+   +----> |  Apply pred. filters
+   |      |
+   |      |                +-------+
+   |      +----+---------->+node 2 |
+   |           |           +--+----+
+   |  watch    |              |
+   |           |              |        +------+
+   |           +---------------------->+node 3|
++--+----------+               |        +--+---+
+| Pods in etcd|               |           |
```
Comment (xiang90) [inline docs/devel/scheduler.md]: change this to Pods in API sever? We want to hide etcd for components outside API server for most the times.

File: docs/devel/scheduler.md [hunk 2]
```diff
@@ -41,45 +41,67 @@ indicating where the Pod should be scheduled.
...
+                                    |
+                                    | node 1: p=2
+                                    | node 2: p=5
+                                    v
+                               select max{node priority} = node 2
+
+```
+
+The Scheduler tries to find a node for each Pod, one at a time.  Notices pods via watch.
+- First it applies a set of "predicates" to filter out inappropriate nodes inappropriate nodes. For example, if the PodSpec specifies resource requests, then the scheduler will filter out nodes that don't have at least that much resources available (computed as the capacity of the node minus the sum of the resource requests of the containers that are already running on the node).
```
Comment (xiang90) [inline docs/devel/scheduler.md]: inappropriate nodes inappropriate nodes -> inappropriate nodes

File: docs/devel/scheduler.md (hunk 1 above)
Comment (timothysc) [inline docs/devel/scheduler.md]: +1

File: docs/devel/scheduler.md [hunk 3]
```diff
@@ -41,45 +41,67 @@ indicating where the Pod should be scheduled.
...
+- Second, it applies a set of "priority functions"
+that rank the nodes that weren't filtered out by the predicate check. For example, it tries to spread Pods across nodes and zones while at the same time favoring the least-loaded nodes (where "load" here is sum of the resource requests of the containers running on the node,
```
Comment (timothysc) [inline docs/devel/scheduler.md]: Might want to call that "theoretical load".

**Comment by jayunit100** [comment]
Comments addressed, separate commit for new improvements

File: docs/devel/scheduler.md [hunk 4]
```diff
@@ -41,45 +41,66 @@ indicating where the Pod should be scheduled.
...
+                                    | node 2: p=5
+                                    v
+                               select max{node priority} = node 2
+
+```
+
+The Scheduler tries to find a node for each Pod, one at a time.  Notices pods via watch.
+- First it applies a set of "predicates" to filter out inappropriate nodes. For example, if the PodSpec specifies resource requests, then the scheduler will filter out nodes that don't have at least that much resources available (computed as the capacity of the node minus the sum of the resource requests of the containers that are already running on the node).
+- Second, it applies a set of "priority functions"
+that rank the nodes that weren't filtered out by the predicate check. For example, it tries to spread Pods across nodes and zones while at the same time favoring the least (theoretically) loaded nodes (where "load" is measured theoretically as the sum of the resource requests of the containers running on the node, divided by the node's capacity).
```
Comment (timothysc) [inline docs/devel/scheduler.md]: weird to say theoretically twice in succession.

File: docs/devel/scheduler.md [hunk 5]
```diff
@@ -41,45 +41,66 @@ indicating where the Pod should be scheduled.
...
+                      +-------------+--------------+
+                                    |
+                                    | node 1: p=2
+                                    | node 2: p=5
+                                    v
+                               select max{node priority} = node 2
+
+```
+
+The Scheduler tries to find a node for each Pod, one at a time.  Notices pods via watch.
```
Comment (timothysc) [inline docs/devel/scheduler.md]: s/Notices pods via watch./ ""

unless you want to elaborate on the details.

**Comment by timothysc** [comment]
minor nits then I think lgtm.

**Comment by jayunit100** [comment]
updated..

**Comment by timothysc** [comment]
lgtm.

**Pull request merged**